### PR TITLE
Add url_for_version function to qt

### PR DESCRIPTION
**Pull request by adamjstewart**
If the qt download directory changes one more time...

I added the most recent version, removed a duplicate import, and turned gtkplus into an optional dependency as decided in #166.

### CODE REVIEW

File: var/spack/repos/builtin/packages/qt/package.py [hunk 1]
```diff
@@ -1,37 +1,29 @@
...
+    url      = 'http://download.qt.io/archive/qt/5.5/5.5.1/single/qt-everywhere-opensource-src-5.5.1.tar.gz'
 
-    version('3.3.8b', '9f05b4125cfe477cc52c9742c3c09009',
-            url="http://download.qt.io/archive/qt/3/qt-x11-free-3.3.8b.tar.gz")
+    version('5.5.1',  '59f0216819152b77536cf660b015d784')
+    version('5.4.0',  'e8654e4b37dd98039ba20da7a53877e6')
+    version('5.3.2',  'febb001129927a70174467ecb508a682')
+    version('5.2.1',  'a78408c887c04c34ce615da690e0b4c8')
+    version('4.8.6',  '2edbe4d6c2eff33ef91732602f3518eb')
+    version('3.3.8b', '9f05b4125cfe477cc52c9742c3c09009')
```
Comment (mathstuf) [inline var/spack/repos/builtin/packages/qt/package.py]: Oh, wow. This should probably just be removed…

File: var/spack/repos/builtin/packages/qt/package.py [hunk 2]
```diff
@@ -52,6 +44,34 @@ class Qt(Package):
...
+        elif version >= Version('4.6'):
+            url += "%s/qt-everywhere-opensource-src-%s.tar.gz" % \
+                    (version.up_to(2), version)
+        elif version >= Version('4.0'):
+            url += "%s/qt-x11-opensource-src-%s.tar.gz" % \
+                    (version.up_to(2), version)
+        elif version >= Version('3'):
+            url += "%s/qt-x11-free-%s.tar.gz" % \
+                    (version.up_to(1), version)
+        elif version >= Version('2.1'):
```
Comment (mathstuf) [inline var/spack/repos/builtin/packages/qt/package.py]: Qt 2 has probably been dead for a long time; no need to waste bytes to any support for it. [Here's](https://en.wikipedia.org/wiki/List_of_Qt_releases) the release history.

**Comment by adamjstewart** [comment]
@mathstuf: I would be fine with only keeping versions 5 and above. It would make url_for_version a lot simpler.

Does anyone else have any thoughts on this? How important is being thorough and supporting older versions vs. keeping life simple?

**Comment by mathstuf** [comment]
Versions in the 4.8 series and above would be fine. 4 is on its way out, but isn't completely gone yet. Qt3 is pretty much dead at this point. Debian only has Qt3 in [oldoldstable](https://packages.debian.org/search?keywords=libqt3), but Qt4 is [still around](https://packages.debian.org/search?keywords=libqt4).

**Comment by tgamblin** [comment]
We need to keep 3 for OpenSpeedShop, but I don't see why we need v2.


________________________________

**Comment by adamjstewart** [comment]
If we're going to keep 5 out of 7 if-statements then we might as well keep all of them just to be thorough.

**Comment by adamjstewart** [comment]
Someone save me. I have no idea how to use Git...

**Comment by tgamblin** [comment]
Usually when this happens I just check out `develop` again and cherry-pick the stuff I want.  I can't really see your branches but you may be able to just rebase your PR on your develop branch, which does mostly the same thing.

**Comment by tgamblin** [comment]
https://xkcd.com/1597/

**Comment by adamjstewart** [comment]
This is so true that I'm not sure if it's sad or funny...

**Comment by tgamblin** [comment]
git is just homeomorphic endofunctors

**Comment by tgamblin** [comment]
Seriously, though, I really like git.  Learn git!

**Comment by adamjstewart** [comment]
Ugh, fine... Know any good tutorials? I already went through this one: https://www.atlassian.com/git/tutorials/

**Comment by tgamblin** [comment]
I really liked [the book](https://progit.org).  Reading the whole thing was probably the most helpful thing for me.  Or at least chapters 1-5.

**Comment by adamjstewart** [comment]
I'll check it out, thanks!

**Pull request closed without merging**